src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1328 @@
+        $this->doubleClickOnElement($this->findElement($xpath));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rightClick(
+        #[Language('XPath')]
+        $xpath
+    ): void {
+        $this->rightClickOnElement($this->findElement($xpath));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function attachFile(
+        #[Language('XPath')]
+        $xpath,
+        #[Language('file-reference')]
+        $path
+    ): void {
+        $element = $this->findElement($xpath);
+        $this->ensureInputType($element, $xpath, 'file', 'attach a file on');
+
+        // @todo - Check if this is the correct way to upload files
+        $element->sendKeys($path);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isVisible(
+        #[Language('XPath')]
+        $xpath
+    ): bool {
+        return $this->findElement($xpath)->isDisplayed();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function mouseOver(
+        #[Language('XPath')]
+        $xpath
+    ): void {
+        $this->mouseOverElement($this->findElement($xpath));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function focus(
+        #[Language('XPath')]
+        $xpath
+    ): void {
+        $this->trigger($xpath, 'focus');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function blur(
+        #[Language('XPath')]
+        $xpath
+    ): void {
+        $this->trigger($xpath, 'blur');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keyPress(
+        #[Language('XPath')]
+        $xpath,
+        $char,
+        $modifier = null
+    ): void {
+        $options = $this->charToSynOptions($char, $modifier);
+        $this->trigger($xpath, 'keypress', $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keyDown(
+        #[Language('XPath')]
+        $xpath,
+        $char,
+        $modifier = null
+    ): void {
+        $options = $this->charToSynOptions($char, $modifier);
+        $this->trigger($xpath, 'keydown', $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keyUp(
+        #[Language('XPath')]
+        $xpath,
+        $char,
+        $modifier = null
+    ): void {
+        $options = $this->charToSynOptions($char, $modifier);
+        $this->trigger($xpath, 'keyup', $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dragTo(
+        #[Language('XPath')]
+        $sourceXpath,
+        #[Language('XPath')]
+        $destinationXpath
+    ): void {
+        $source = $this->findElement($sourceXpath);
+        $destination = $this->findElement($destinationXpath);
+        $this->getWebDriver()->action()->dragAndDrop($source, $destination)->perform();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function executeScript(
+        #[Language('JavaScript')]
+        $script
+    ): void {
+        if (preg_match('/^function[\s(]/', $script ?? '')) {
```

Comment:
`?? ''` is useless. Null values are not supported in this method for the argument